packages/php-wasm/universal/src/lib/php-process-manager.ts

```diff
@@ -160,36 +160,38 @@ export class PHPProcessManager implements AsyncDisposable {
 			await this.getPrimaryPhp();
 		}
 
+		let spawnedPhp: SpawnedPHP;
+
 		if (this.primaryIdle && considerPrimary) {
 			this.primaryIdle = false;
-			return {
+			spawnedPhp = {
 				php: await this.getPrimaryPhp(),
 				reap: () => {
 					this.primaryIdle = true;
 				},
 			};
-		}
-
-		/**
-		 * nextInstance is null:
-		 *
-		 * * Before the first concurrent getInstance() call
-		 * * When the last getInstance() call did not have enough
-		 *   budget left to optimistically start spawning the next
-		 *   instance.
-		 */
-		const spawnedPhp =
-			this.nextInstance || this.spawn({ isPrimary: false });
-
-		/**
-		 * Start spawning the next instance if there's still room. We can't
-		 * just always spawn the next instance because spawn() can fail
-		 * asynchronously and then we'll get an unhandled promise rejection.
-		 */
-		if (this.semaphore.remaining > 0) {
-			this.nextInstance = this.spawn({ isPrimary: false });
 		} else {
-			this.nextInstance = null;
+			/**
+			 * nextInstance is null:
+			 *
+			 * * Before the first concurrent getInstance() call
+			 * * When the last getInstance() call did not have enough
+			 *   budget left to optimistically start spawning the next
+			 *   instance.
+			 */
+			spawnedPhp = await (this.nextInstance ||
+				this.spawn({ isPrimary: false }));
+
+			/**
+			 * Start spawning the next instance if there's still room. We can't
+			 * just always spawn the next instance because spawn() can fail
+			 * asynchronously and then we'll get an unhandled promise rejection.
+			 */
+			if (this.semaphore.remaining > 0) {
+				this.nextInstance = this.spawn({ isPrimary: false });
+			} else {
+				this.nextInstance = null;
+			}
 		}
 		return await spawnedPhp;
 	}
```

Comment:
Are the change in this file actually changing the logic? Or can they be rolled back?

Reply:
I didn't get to cleaning up the PR before you reviewed. I think these are useless changes left from previous exploration within the PR. They can be rolled back.